scripts/zones/Apollyon/mobs/Dee_Wapa_the_Desolator.lua

```diff
@@ -2,52 +2,81 @@
 -- Area: Apollyon CS
 --  Mob: Dee Wapa the Desolator
 -----------------------------------
-require("scripts/globals/limbus");
------------------------------------
+local ID = require("scripts/zones/Apollyon/IDs")
 
 function onMobSpawn(mob)
-    mob:setMobMod(tpz.mobMod.SUPERLINK, mob:getShortID());
-end;
-
-function onMobEngaged(mob,target)
-    local mobID = mob:getID();
-    local X = mob:getXPos();
-    local Y = mob:getYPos();
-    local Z = mob:getZPos();
-    SpawnMob(16933148):setMobMod(tpz.mobMod.SUPERLINK, mob:getShortID());
-    SpawnMob(16933147):setMobMod(tpz.mobMod.SUPERLINK, mob:getShortID());
-    SpawnMob(16933149):setMobMod(tpz.mobMod.SUPERLINK, mob:getShortID());
-    SpawnMob(16933146):setMobMod(tpz.mobMod.SUPERLINK, mob:getShortID());
-end;
-
-function onMobFight(mob,target)
-    local mobID = mob:getID();
-    local X = mob:getXPos();
-    local Y = mob:getYPos();
-    local Z = mob:getZPos();
-    local lifepourcent= ((mob:getHP()/mob:getMaxHP())*100);
-    local instancetime = target:getBattlefieldTimeLeft(Central_Temenos_2nd_Floor);
+    mob:setMobMod(tpz.mobMod.SUPERLINK, mob:getShortID())
+end
 
-    if (lifepourcent < 50 and GetNPCByID(16933247):getAnimation() == 8) then
-        SpawnMob(16933151):setMobMod(tpz.mobMod.SUPERLINK, mob:getShortID());
-        SpawnMob(16933150):setMobMod(tpz.mobMod.SUPERLINK, mob:getShortID());
-        SpawnMob(16933152):setMobMod(tpz.mobMod.SUPERLINK, mob:getShortID());
-        GetNPCByID(16933247):setAnimation(9);
+function onMobEngaged(mob, target)
+    local battlefield = mob:getBattlefield()
+    if battlefield:getLocalVar("startTime") == 0 then
+        battlefield:setLocalVar("startTime", battlefield:getRemainingTime())
     end
+    SpawnMob(ID.mob.APOLLYON_CS_MOB[3]+3):setMobMod(tpz.mobMod.SUPERLINK, mob:getShortID())
+    SpawnMob(ID.mob.APOLLYON_CS_MOB[3]+4):setMobMod(tpz.mobMod.SUPERLINK, mob:getShortID())
+    SpawnMob(ID.mob.APOLLYON_CS_MOB[3]+5):setMobMod(tpz.mobMod.SUPERLINK, mob:getShortID())
+    mob:setLocalVar("wave", 1)
```

Comment:
I believe `onMobEngaged` gets called again upon successive engages after wipes. So may want to leverage the sets for the wave behavior (will have to check current wave var for the race) with a `limbus.allDead` call to make sure you're not double-spawning a wave after a wipe.